perfkitbenchmarker/providers/yandexcloud/yc_virtual_machine.py

```diff
@@ -0,0 +1,475 @@
+    self.network = yc_network.YcNetwork.GetNetwork(self)
+    self.boot_disk_size = vm_spec.boot_disk_size
+    self.boot_disk_type = vm_spec.boot_disk_type
+    self.id = None
+
+    if not self.cores or not self.memory:
+        raise errors.Config.MissingOption
+
+    if self.preemptible and not self.core_fraction:
+        raise errors.Config.MissingOption
+
+  @property
+  def host_list(self):
+    """Returns the list of hosts that are compatible with this VM."""
+    return self.host_map[(self.folder_id, self.zone)]
+
+  def _GenerateCreateCommand(self, public_key):
+    """Generates a command to create the VM instance.
+
+    Args:
+      ssh_keys_path: string. Path to a file containing the ssh-keys metadata.
+
+    Returns:
+      YcCommand. yc command to issue in order to create the VM instance.
+    """
+    boot_disk = {
+        'auto-delete': True
+    }
+    boot_disk['name'] = self.name
+    boot_disk['size'] = self.boot_disk_size or self.BOOT_DISK_SIZE_GB
+    boot_disk['type'] = self.boot_disk_type or self.BOOT_DISK_TYPE
+    if self.image:
+      boot_disk['image-name'] = self.image_name
+    if self.image_family:
+      boot_disk['image-family'] = self.image_family
+    if self.image_folder_id:
+      boot_disk['image-folder-id'] = self.image_folder_id
+
+    args = []
+    args.extend(['compute', 'instances', 'create'])
+
+    cmd = util.YcCommand(self, *args)
+    cmd.flags['name'] = self.name
+    cmd.flags['network-interface'] = util.FormatTags({
+        'subnet-name': self.network.subnet_resource.name,
+        'nat-ip-version': 'ipv4',
+    })
+    if boot_disk:
+      cmd.flags['create-boot-disk'] = util.FormatTags(boot_disk)
+    if self.platform_id:
+      cmd.flags['platform-id'] = self.platform_id
+    cmd.flags['cores'] = self.cores
+    cmd.flags['memory'] = self.memory
+    cmd.flags['zone'] = self.zone
+    cmd.flags['ssh-key'] = public_key
+
+    metadata_from_file = flag_util.ParseKeyValuePairs(
+        FLAGS.yc_instance_metadata_from_file)
+    cmd.flags['metadata-from-file'] = util.FormatTags(metadata_from_file)
+
+    metadata = {'owner': FLAGS.owner} if FLAGS.owner else {}
+    metadata.update(self.boot_metadata)
+    parsed_metadata = flag_util.ParseKeyValuePairs(FLAGS.yc_instance_metadata)
+    for key, value in parsed_metadata.items():
+      if key in metadata:
+        logging.warning('Metadata "%s" is set internally. Cannot be overridden '
+                        'from command line.', key)
+        continue
+      metadata[key] = value
+    cmd.flags['metadata'] = util.FormatTags(metadata)
+
+    if self.preemptible:
+      cmd.flags['preemptible'] = True
+    return cmd
+
+  def _Create(self):
+    """Create a YC VM instance."""
+    with open(self.ssh_public_key) as f:
+      public_key = f.read().rstrip('\n')
+    with vm_util.NamedTemporaryFile(mode='w', dir=vm_util.GetTempDir(),
+                                    prefix='key-metadata') as tf:
+      tf.write(public_key)
+      tf.close()
```

Comment:
Seems a little odd to just read in a file and spit it back out.  Can you just use the self.ssh_public_key in the call to GenerateCreateCommand?